File: Algorithm/Python/zigzag_conversion.py
class Solution:
    def convert(self,s,nRows):
        if nRows==1:
            return s
        tmp=['' for x in range(nRows)]
        position=-1
        direction=1
        for i in range(len(s)):
            position+=direction
            if position==0:
                direction=1
            elif position==nRows-1:
                direction=-1
            tmp[position]+=s[i]
        return ''.join(tmp)
if __name__=='__main__':
    sol=Solution()
    s='PAYPALISHIRING'
    nRows=3
    print sol.convert(s,nRows)

File: Algorithm/CPP/zigzag_conversion.cpp
# include<stdio.h>
# include <iostream>
# include <string>
# include <array>
# include <vector>
using namespace std;
class Solution {
public:
    string convert(string s, int numRows) {
	 if (numRows == 1) return s;
		int position=-1, direction = 1;
  
		vector<string> vec(numRows);
    	for ( auto chr:s)
    	{
    		position += direction;
    		if (position == 0)
			{
				direction = 1;
			}
			else if (position == numRows-1)
			{
				direction = -1;
			}
			vec[position] += chr;
		}
		string ans;
		for (auto s:vec)  ans += s;
		return ans;
    }
};
int main()
{
	Solution sol;
	string s = "PAYPALISHIRING";
	int numRows = 3;

	cout<< sol.convert(s,numRows);

    return 0;
}

